src/udp/connection/encoded_id.rs

```diff
@@ -0,0 +1,346 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::Hasher;
+use std::time::Duration;
+
+use blowfish::Blowfish;
+use byteorder::{LittleEndian, LE};
+use cipher::generic_array::GenericArray;
+use cipher::{BlockDecrypt, BlockEncrypt, BlockSizeUser};
+
+use crate::keys::{BLOWFISH_KEY, TEST_BLOWFISH_KEY};
+use crate::protocol::clock::{Clock, LocalClock};
+
+type BlowfishArray = GenericArray<u8, <Blowfish<LittleEndian> as BlockSizeUser>::BlockSize>;
+
+use super::client_id::{HashedId, KeyedClientId, PlainClientId};
+
+pub trait EncodedId<T: HashedId> {
```

Comment:
@da2ce7 I like the name `EncodedId`. In my case I used `ConenctionIdData` becuase:

- It's not encoded. You have to encode it by calling `to_bytes()`. Maybe a better name would be `connection_id_data.encode()`.
- I wanted to make it more explicit that this is not the final connection id, it's the data you are going to put inside de connection id. In fact, you need to encode it again into an i64.

Reply:
> @da2ce7 I like the name `EncodedId`. In my case I used `ConenctionIdData` becuase:
> 
>     * It's not encoded. You have to encode it by calling `to_bytes()`. Maybe a better name would be `connection_id_data.encode()`.
> 
>     * I wanted to make it more explicit that this is not the final connection id, it's the data you are going to put inside de connection id. In fact, you need to encode it again into an i64.

I have thought about the names, and I have come up with:

`ClientId -> ClientImage`
`ConnectionId -> ConnectionCookie`

Because the Id is really just a temporary lifetime cookie that we give to the clients.